pkg/logs/input/docker/tailer.go

```diff
@@ -36,53 +36,57 @@ const tagsUpdatePeriod = 10 * time.Second
 // Logs from stdout and stderr are multiplexed into a single channel and needs to be demultiplexed later one.
 // To multiplex logs, docker adds a header to all logs with format '[SEV][TS] [MSG]'.
 type Tailer struct {
-	ContainerID   string
-	outputChan    chan *message.Message
-	decoder       *decoder.Decoder
-	reader        io.ReadCloser
-	cli           *client.Client
-	source        *config.LogSource
-	containerTags []string
+	container  *Container
+	outputChan chan *message.Message
+	decoder    *decoder.Decoder
+	reader     io.ReadCloser
+	cli        *client.Client
+	source     *config.LogSource
+	// containersSource is the source that uses short image name as source and service
+	// tag if applicable (i.e. no k8s annotation nor docker label)
+	containerSource *config.LogSource
+	containerTags   []string
 
-	sleepDuration      time.Duration
-	shouldStop         bool
-	stop               chan struct{}
-	done               chan struct{}
-	erroredContainerID chan string
-	cancelFunc         context.CancelFunc
-	lastSince          string
-	mutex              sync.Mutex
+	sleepDuration    time.Duration
+	shouldStop       bool
+	stop             chan struct{}
+	done             chan struct{}
+	erroredContainer chan *Container
+	cancelFunc       context.CancelFunc
+	lastSince        string
+	mutex            sync.Mutex
 }
 
 // NewTailer returns a new Tailer
-func NewTailer(cli *client.Client, containerID string, source *config.LogSource, outputChan chan *message.Message, erroredContainerID chan string) *Tailer {
+func NewTailer(cli *client.Client, container *Container, source *config.LogSource, outputChan chan *message.Message, erroredContainer chan *Container) *Tailer {
 	return &Tailer{
-		ContainerID:        containerID,
-		outputChan:         outputChan,
-		decoder:            decoder.InitializeDecoder(source, dockerParser),
-		source:             source,
-		cli:                cli,
-		sleepDuration:      defaultSleepDuration,
-		stop:               make(chan struct{}, 1),
-		done:               make(chan struct{}, 1),
-		erroredContainerID: erroredContainerID,
+		container:        container,
+		outputChan:       outputChan,
+		decoder:          decoder.InitializeDecoder(source, dockerParser),
+		source:           source,
+		containerSource:  container.GetSourceShortImageName(source),
```

Comment:
I think it's messy to initialize the `containerSource` here, can we inject it upstream, i.e. in the `Launcher` ?